### PR TITLE
wallet connect integration

### PR DESCRIPTION
**Pull request by marshacb**
Adds support for wallet connect to the polkadot staking dashboard
* provides option for wallet connect in the extensions dropdown
* when selected opens web3modal which allows for wallet session creation/pairing using a qrcode or by copying and providing a generated session URI

### CODE REVIEW

**Comment by rossbulat** [comment]
Hey @marshacb, this is fantastic. I aim to dedicate some real time to get this merged. 

Something for the future may be to move Wallet Connect to a standalone component in a UI library repo I am building in the background, it is at https://github.com/paritytech/polkadot-dashboard-ui. Top priority would be to get this PR working well in the dashboard firstly, then in a later update move it to a separate component that all Polkadot apps can leverage.

**Comment by marshacb** [comment]
> Hey @marshacb, this is fantastic. I aim to dedicate some real time to get this merged.
> 
> Something for the future may be to move Wallet Connect to a standalone component in a UI library repo I am building in the background, it is at https://github.com/paritytech/polkadot-dashboard-ui. Top priority would be to get this PR working well in the dashboard firstly, then in a later update move it to a separate component that all Polkadot apps can leverage.

Thanks @rossbulat! Making this into a standalone component that could be used by many Polkadot eco apps sounds awesome! I tried to make things as simple as possible for this but I'd definitely appreciate your feedback on things that might be changed or improved.

**Comment by rossbulat** [comment]
⚠️ master -> main branch migration ⚠️
Polkadot staking dashboard's master branch has been renamed to main. If developing locally, remember to update .git/config and fetch main to get the latest default branch.

**Comment by rossbulat** [comment]
Hi @marshacb, I'm going to spend some time refactoring a couple of things if this is okay to do on this branch?

File: src/contexts/Connect/index.tsx [hunk 1]
```diff
@@ -368,6 +499,103 @@ export const ConnectProvider = ({
       updateInitialisedExtensions(
         `unknown_extension_${extensionsInitialisedRef.current.length + 1}`
       );
+    } else if (id === 'wallet-connect') {
+      const provider = await WalletConnectInitializer.initialize();
```
Comment (rossbulat) [inline src/contexts/Connect/index.tsx:503]: @marshacb Do we only need to do this if there is not an existing provider assigned to `wcChainInfo`?

**Comment by rossbulat** [comment]
I am finding it challenging rebasing this PR as there have been some structural changes to Connect and Extensions. I think manually committing some files will make it easier to rebase once the code is more inline. Will see what I can do.

File: src/contexts/Connect/index.tsx (hunk 1 above)
Comment (marshacb) [reply src/contexts/Connect/index.tsx:503]: @rossbulat yes, if there is already a provider/session assigned in the state we don't need to re-initialize one again

Comment (rossbulat) [reply src/contexts/Connect/index.tsx:503]: Cool okay,I'll explore when we can cut down on code. 👍

**Comment by rossbulat** [comment]
@marshacb Do you know if there are any apps we can use to test Wallet Connect with currently?

File: src/styles/Home.module.css [hunk 2]
```diff
@@ -0,0 +1,152 @@
+.container {
```
Comment (rossbulat) [inline src/styles/Home.module.css:1]: Where are these styles applied? Even with this file deleted I cannot see any changes to the Wallet Connect modal that pops up. Disclaimer - I haven't tried importing any accounts yet.

**Comment by marshacb** [comment]
> @marshacb Do you know if there are any apps we can use to test Wallet Connect with currently?

@rossbulat yes, you can use wallet connects test wallet to connect using the wallet connect modal. Its found here https://react-wallet.walletconnect.com/walletconnect

**Comment by rossbulat** [comment]
> > @marshacb Do you know if there are any apps we can use to test Wallet Connect with currently?
> 
> 
> 
> @rossbulat yes, you can use wallet connects test wallet to connect using the wallet connect modal. Its found here https://react-wallet.walletconnect.com/walletconnect

Awesome, I'll check it out. Thanks!

File: src/styles/Home.module.css (hunk 2 above)
Comment (marshacb) [reply src/styles/Home.module.css:1]: @rossbulat ahhh, I didn't intend on making any style changes in this PR so these can be deleted

**Comment by marshacb** [comment]
> > > @marshacb Do you know if there are any apps we can use to test Wallet Connect with currently?
> > 
> > 
> > @rossbulat yes, you can use wallet connects test wallet to connect using the wallet connect modal. Its found here https://react-wallet.walletconnect.com/walletconnect
> 
> Awesome, I'll check it out. Thanks!

@rossbulat Awesome! Also, if you want to use this branch to test in relation to your rebased branch I've updated it.

**Pull request closed without merging**